pkg/controller/config/config_controller.go

```diff
@@ -273,25 +283,28 @@ func (r *ReconcileConfig) Reconcile(ctx context.Context, request reconcile.Reque
 	if r.needsReplay == nil {
 		r.needsReplay = r.watched.Intersection(newSyncOnly)
 	}
+
+	// Wipe all data to avoid stale state if needed. Happens once per watch-set-change.
+	if err := r.wipeCacheIfNeeded(ctx); err != nil {
+		return reconcile.Result{}, fmt.Errorf("wiping opa data cache: %w", err)
+	}
+
+	// TODO: lock watched
```

Comment:
Do we need to satisfy these TODOs to avoid a race condition with audit from cache?

Reply:
No. `Set` handles locking correctly.